src/Core/Domain/Cart/QueryHandler/GetCartSummaryForOrderCreationHandler.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Cart\QueryHandler;
+
+use PrestaShop\PrestaShop\Core\Domain\Cart\Command\CreateEmptyCustomerCartCommand;
+use PrestaShop\PrestaShop\Core\Domain\Cart\CommandHandler\CreateEmptyCustomerCartHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Cart\Query\GetCartSummary;
+use PrestaShop\PrestaShop\Core\Domain\Cart\Query\GetCartSummaryForOrderCreation;
+use PrestaShop\PrestaShop\Core\Domain\Cart\Query\GetLastEmptyCustomerCart;
+
+final class GetCartSummaryForOrderCreationHandler implements GetCartSummaryForOrderCreationHandlerInterface
+{
+    /**
+     * @var GetLastEmptyCustomerCartHandlerInterface
+     */
+    private $getLastEmptyCustomerCartHandler;
+
+    /**
+     * @var CreateEmptyCustomerCartHandlerInterface
+     */
+    private $createEmptyCartForCustomerHandler;
+
+    /**
+     * @var GetCartSummaryHandlerInterface
+     */
+    private $getCartSummaryHandler;
+
+    public function __construct(
+        GetLastEmptyCustomerCartHandlerInterface $getLastEmptyCustomerCartHandler,
+        CreateEmptyCustomerCartHandlerInterface $createEmptyCartForCustomerHandler,
+        GetCartSummaryHandlerInterface $getCartSummaryHandler
```

Comment:
i chose to inject handlers because injecting command/query buses feels like injecting service container. Of course, if we were to have any middlewares, those wouldnt be executed for injected handlers. wdyt about it?

Reply:
1) I 👍 injecting handlers, you're right about not injecting Bus

2) The naming seems OK to me. We can always look for better, but this looks good enough.

3) If we end up writing spaghetti code because the code does a lot of things, it's because the feature is complex.
I think handlers can **use** other handlers but not dispatch more commands. However if we have more than 2 level of calls (handler 1 uses handler 2 uses handler 3 uses handler 4) we have to question ourselves whether what we are trying to achieve is right.

I think 1 level call (handler 1 uses handler 2) is OK . 2 level calls is only for complex scenarii (and order management IS a complex scenario). but 3 level calls ... OK there's something that we need tor refactor first.